### PR TITLE
Odyssey: Add opt-out notice for odyssey

### PR DESCRIPTION
**Pull request by kangzj**
#### Proposed Changes

* Adds`packages/components/src/notice-banner` which is forked from [Notice](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/components/notice) adjusted to use Calypso standard
* Adds Opt-out notice to Traffic page highlight section
* Adds supports to access existing `jetpack/v4` API directly
* Sends dismissing notice request to JITM endpoint for the site to remember the notice is dismissed

#### Testing Instructions

- Use the branch from https://github.com/Automattic/jetpack/pull/28733 for Jetpack the plugin
- Setup Odyssey with Option 1 in `PejTkB-3E-p2` <- enable New Stats if not yet enabled
- Open `/wp-admin/admin.php?page=stats`
- Ensure an opt-out notice is shown on the Traffic page
- Click `manage your settings`
- Ensure you are taken to Jetpack Stats settings section
- Return to `/wp-admin/admin.php?page=stats`
- Click `x`
- Ensure the notice is dismissed
- Refresh the page
- Ensure the notice wouldn't appear again

image

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #72736

### CODE REVIEW

File: client/lib/wpcom-xhr-wrapper/index.js [hunk 1]
```diff
@@ -27,7 +27,8 @@ export async function jetpack_site_xhr_wrapper( params, callback ) {
 			'X-WP-Nonce': config( 'nonce' ),
 		},
 		isRestAPI: false,
-		apiNamespace: 'jetpack/v4/stats-app',
+		apiNamespace:
+			params.apiNamespace === 'jetpack/v4' ? params.apiNamespace : 'jetpack/v4/stats-app',
```
Comment (kangzj) [inline client/lib/wpcom-xhr-wrapper/index.js:31]: If the API is `jetpack/v4` then we don't need to prefix it with `stats-app`, which are only for compensated APIs. This allows Odyssey to access existing `jetpack/v4` APIs, but will probably need nonce etc to function in some cases.

Comment (kangzj) [reply client/lib/wpcom-xhr-wrapper/index.js:31]: The trick to call an existing `jetpack/v4` API is to set `apiNamespace` to `jetpack/v4`

File: client/state/sites/selectors/has-opt-out-new-stats-notice.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * @param siteId The site ID.
+ * @returns {?boolean}        hasOptOutNotice
+ */
+export default function hasOptOutNewStatsNotice( state, siteId ) {
+	if ( ! siteId ) {
+		return null;
+	}
+	const site = getRawSite( state, siteId );
+
+	return site?.has_opt_out_new_stats_notice;
```
Comment (dognose24) [inline client/state/sites/selectors/has-opt-out-new-stats-notice.js]:
```suggestion
	return site?.stats_notices?.opt_out_new_stats;
```

I am not sure about the correctness, but I have to change the site config properties to make the notice show.

截圖 2023-02-03 下午3 28 55

I found the [returned site config](https://github.com/Automattic/jetpack/pull/28733/files#diff-f2c1a159814b020c8ba57e8123158c5d50c2d078c14b96de2da89fde1d89cf74R252) from the dependent Jetpack changes has the object structure corresponding to my suggestion above.

Comment (kangzj) [reply client/state/sites/selectors/has-opt-out-new-stats-notice.js]: Good catch. I didn't push the changes to remote oops 😄 fixed now. Thanks

File: client/state/data-layer/wpcom/sites/jitm/index.js [hunk 3]
```diff
@@ -97,6 +97,27 @@ export const doDismissJITM = ( action ) =>
 		action
 	);
 
+/**
+ * Dismisses a jitm on the jetpack site, it returns nothing useful and will return no useful error, so we'll
```
Comment (dognose24) [inline client/state/data-layer/wpcom/sites/jitm/index.js]: Could we explain the `doDismissJITMDirect` in more detail to identify the difference from `doDismissJITM`?

**Review by dognose24 — APPROVED** [review]
Works as described! I am still determining the annotation suggestion; please feel free to ship without updating it if unnecessary.

File: client/state/data-layer/wpcom/sites/jitm/index.js (hunk 3 above)
Comment (kangzj) [reply client/state/data-layer/wpcom/sites/jitm/index.js]: Good point. Added comment in 75d9257

**Review by dognose24 — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7798671

Thank you @kangzj for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7798671).

**Pull request merged**